### PR TITLE
feat: add feature flag for multi-provider configuration

### PR DESCRIPTION
**Pull request by davidsoderberg**
### What change does this PR introduce?
New feature flag for the multi-provider configuration.

### Why was this change needed?
To disable logic for multi-provider configuration until feature is ready.

### CODE REVIEW

File: apps/api/src/app/feature-flags/use-cases/get-feature-flag/get-feature-flag.use-case.ts [hunk 1]
```diff
@@ -1,5 +1,4 @@
-import { Injectable, Logger } from '@nestjs/common';
-import { EnvironmentWithUserCommand } from '@novu/application-generic';
```
Comment (davidsoderberg) [inline apps/api/src/app/feature-flags/use-cases/get-feature-flag/get-feature-flag.use-case.ts:2]: Removing unused imports.

File: libs/shared/src/types/feature-flags/index.ts [hunk 2]
```diff
@@ -1,4 +1,5 @@
 export enum FeatureFlagsKeysEnum {
   IS_TEMPLATE_STORE_ENABLED = 'IS_TEMPLATE_STORE_ENABLED',
   IS_TOPIC_NOTIFICATION_ENABLED = 'IS_TOPIC_NOTIFICATION_ENABLED',
+  IS_MULTI_PROVIDER_CONFIGURATION = 'IS_MULTI_PROVIDER_CONFIGURATION',
```
Comment (p-fernandez) [inline libs/shared/src/types/feature-flags/index.ts:4]:
```suggestion
  IS_MULTI_PROVIDER_CONFIGURATION_ENABLED = 'IS_MULTI_PROVIDER_CONFIGURATION_ENABLED',
```
For consistency.

File: apps/web/src/hooks/useFeatureFlags.ts [hunk 3]
```diff
@@ -19,6 +19,16 @@ export const useIsTemplateStoreEnabled = (): boolean => {
...
 
+export const useIsMultiProviderConfigurationEnabled = (): boolean => {
+  const value = IS_MULTI_PROVIDER_CONFIGURATION;
+  const fallbackValue = false;
+  const defaultValue = prepareBooleanStringFeatureFlag(value, fallbackValue);
+
+  const isTemplateStoreEnabled = useGetFlagByKey<boolean>(FeatureFlagsKeysEnum.IS_MULTI_PROVIDER_CONFIGURATION);
+
+  return isTemplateStoreEnabled ?? defaultValue;
+};
```
Comment (p-fernandez) [inline apps/web/src/hooks/useFeatureFlags.ts:30]:
```suggestion
export const useIsMultiProviderConfigurationEnabled = (): boolean => {
  const value = IS_MULTI_PROVIDER_CONFIGURATION_ENABLED;
  const fallbackValue = false;
  const defaultValue = prepareBooleanStringFeatureFlag(value, fallbackValue);

  const isMultiProviderConfigurationEnabled = useGetFlagByKey<boolean>(FeatureFlagsKeysEnum.IS_MULTI_PROVIDER_CONFIGURATION_ENABLED);

  return isMultiProviderConfigurationEnabled ?? defaultValue;
};
```
Naming update and adding changes from my other suggestion.

File: apps/api/src/app/feature-flags/e2e/get-feature-flag.e2e.ts
Comment (p-fernandez) [inline apps/api/src/app/feature-flags/e2e/get-feature-flag.e2e.ts:1]: We should also add the test that checks E2E with Launch Darkly server, like the ones listed under the describe bit `SDK key environment variable is set`. Even if they are being skipped in the CI for the flakiness, implement it and test it in local. There won't be flakey and we will double check it works. Once the test is checked and passed in the local we can disable in Launch Darkly the feature flag to false as the test won't run in the CI. But will be left done for the future. 👍🏻

**Review by p-fernandez — CHANGES_REQUESTED** [review]

**Review by LetItRock — APPROVED** [review]

**Pull request merged**